From 355df48cfaceeaebf110fd71a1dc1fe668d44792 Mon Sep 17 00:00:00 2001
From: Ashwin <ashwin.bhat@broadcom.com>
Date: Thu, 14 Aug 2014 16:57:27 -0700
Subject: [PATCH 56/77] net: wireless: bcmdhd: Fix pkt filter for 64-bit host

Pkt filters were not being correctly set in 64 bit architecture
because a pointer was defined in union being passed to FW
(pointer is not used by HOST) resulting in different offset to
the next field of struct in HOST and FW.

Bug: 17006509

Change-Id: I2b2b0dec625b1e62329ff475067d010e05d25201
Signed-off-by: Ashwin <ashwin.bhat@broadcom.com>
Signed-off-by: Dmitry Shmidt <dimitrysh@google.com>
---
 drivers/net/wireless/bcmdhd/include/wlioctl.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/net/wireless/bcmdhd/include/wlioctl.h b/drivers/net/wireless/bcmdhd/include/wlioctl.h
index 47c6d7c6123e..cdf89ae83e70 100644
--- a/drivers/net/wireless/bcmdhd/include/wlioctl.h
+++ b/drivers/net/wireless/bcmdhd/include/wlioctl.h
@@ -3010,7 +3010,6 @@ typedef struct wl_pkt_filter_pattern {
 		uint32	offset;		/* Offset within received packet to start pattern matching.
 				 * Offset '0' is the first byte of the ethernet header.
 				 */
-		wl_pkt_decrypter_t*	decrypt_ctx;	/* Decrypt context */
 	};
 	uint32	size_bytes;	/* Size of the pattern.  Bitmask must be the same size. */
 	uint8   mask_and_pattern[1]; /* Variable length mask and pattern data.  mask starts
-- 
2.0.3

